### DISCUSSION TITLE
Invalidate browser cache for site migrated from React to Next.js

### DISCUSSION THREAD

**Discussion by zvitez** [vercel/next.js#56492] [Help]
### Summary

Hi! I migrated a site from React to Next.js and moved it from Netlify to Vercel. In the react site the cache was set to 365 days, so all old visitors to the domain keep seeing the old React website. I'm really agonizing over finding a way to force the browser to clear the cache...I tried redirecting from the domain.com to www.domain.com but the browser keeps loading domain.com.

It seems like I have to do something on the server/Vercel level? I'm still using the Pages folder structure, and tried the following:
```
import type { NextApiRequest, NextApiResponse } from 'next';
 
export const config = {
  runtime: 'edge',
};
 
export default function handler(
  request: NextApiRequest,
  response: NextApiResponse,
) {
  response.setHeader('Cache-Control', 'public, s-maxage=1');
 
  return response.status(200).json({ name: 'John Doe' });
}
```

It didn't help.

I also tried to add the following function to _app and call it from useEffect

```
const caching = () => {
    let version = localStorage.getItem("version");
    if (version != packageJson.version) {
      if ("caches" in window) {
        caches.keys().then((names) => {
          // Delete all the cache files
          names.forEach((name) => {
            caches.delete(name);
          });
        });

        // Makes sure the page reloads. Changes are only visible after you refresh.
        window.location.reload(true);
      }

      localStorage.clear();
      localStorage.setItem("version", packageJson.version);
    }
  };
```

  
I'd appreciate some help!

**Comment by ameeetgaikwad** [comment]
let me know if you find any answers @zvitez
